src/Ship.js

```diff
@@ -0,0 +1,21 @@
+function Ship(itinerary) {
+  this.itinerary = itinerary;
+  this._itineraryPortPosition = 0;
+  this.currentPort = this.itinerary.ports[this._itineraryPortPosition];
+  this.currentPort.addShip(this);
+  this.previousPort = null;
+}
+
+Ship.prototype.setSail = function () {
+  this.previousPort = this.currentPort;
+  this.currentPort.removeShip(this);
+  this.currentPort = null;
+};
+
+Ship.prototype.dock = function (port) {
+  this._itineraryPortPosition++;
```

Comment:
I see what you're doing here by assigning this to a variable and then increasing each time your dock. However, I would suggest using `indexOf` with your `previousPort`, as mentioned in the track, to get the port position. This is because you wouldn't have to worry about accidentally assigning the wrong value to this variable or forgetting to increment it. It doesn't make too much of a difference on a project of this size, but if it were to grow in scale it's just another thing you would then have to keep in mind and could lead to unexpected errors.

If you did want to act like a variable, you could use a getter, similar to:
```
Ship.prototype = {
  get itineraryPortPosition() {
    const ports = this.itinerary.ports;
    return this.currentPort ? ports.indexOf(this.currentPort) : ports.indexOf(this.previousPort) + 1;
  }
}
``` 
This works by checking if the `currentPort` is `truthy`. If it is, it returns the `currentPort` in `itinerary.ports`. If it isn't, it returns the `index` of the `previousPort` in `itinerary.ports` + 1. Though this would involve some more refactoring elsewhere to get working